applications/GeoMechanicsApplication/tests/cpp_tests/test_reset_displacement_process.cpp

```diff
@@ -0,0 +1,158 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_processes/reset_displacement_process.h"
+#include "geo_mechanics_fast_suite.h"
+
+#include <boost/numeric/ublas/assignment.hpp>
+
+namespace
+{
+
+using namespace Kratos;
+
+class StubConstitutiveLaw : public ConstitutiveLaw
+{
+};
+
+class StubElement : public Element
+{
+public:
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(StubElement);
+
+    StubElement(IndexType NewId, const GeometryType::Pointer& pGeometry) : Element(NewId, pGeometry)
+    {
+        mConstitutiveLaws = std::vector<ConstitutiveLaw::Pointer>(3, make_shared<StubConstitutiveLaw>());
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<ConstitutiveLaw::Pointer>& rVariable,
+                                      std::vector<ConstitutiveLaw::Pointer>&    rOutput,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        rOutput = mConstitutiveLaws;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Vector>& rVariable,
+                                      std::vector<Vector>&    rOutput,
+                                      const ProcessInfo&      rCurrentProcessInfo) override
+    {
+        rOutput = mIntegrationPointVectors;
+    }
+
+    using Element::CalculateOnIntegrationPoints;
+
+    std::vector<Vector>                   mIntegrationPointVectors = {};
+    std::vector<ConstitutiveLaw::Pointer> mConstitutiveLaws        = {};
+};
+
+ModelPart& CreateModelPartWithAStubElement(Model& rModel)
+{
+    auto& model_part = rModel.CreateModelPart("MainModelPart");
+    auto  node_1     = model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    auto  node_2     = model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    auto  node_3     = model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+
+    auto geometry = std::make_shared<Triangle2D3<Node>>(node_1, node_2, node_3);
+    auto element  = Kratos::make_intrusive<StubElement>(1, geometry);
+    model_part.AddElement(element);
+
+    return model_part;
+}
+} // namespace
+
+namespace Kratos::Testing
+{
+KRATOS_TEST_CASE_IN_SUITE(ResetDisplacementProcess_SetsInitialStressOfConstitutiveLaws,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
```

Comment:
I was a bit in two minds here: on the one hand, I don't want to couple the tests to implementation details (e.g. I don't want to change the test when we would decide to use cauchy stress,. However, from a 'test as documentation' point of view, it might help to see here that it is required that the element can return the PK2_STRESS_VECTOR.

Since we now use SetValuesOnIntegrationPoints and have to input a stress measure there anyway, I think it's best to just be explicit, but I think it's still something to keep in mind, to not have the tests depend on implementation details